x-pack/plugins/apm/public/components/shared/charts/timeseries_chart.tsx

```diff
@@ -118,19 +110,46 @@ export function TimeseriesChart({
       ? anomalyChartTimeseries?.boundaries
       : []),
     ...(anomalyChartTimeseries?.scores ?? []),
-  ];
+  ]
+    // Sorting series so that area type series are before line series
+    // This is a workaround so that the legendSort works correctly
+    .sort(
+      isComparingExpectedBounds
+        ? (prev, curr) => prev.type.localeCompare(curr.type)
+        : undefined
+    );
+
+  const xValues = timeseries.flatMap(({ data }) => data.map(({ x }) => x));
+
+  const timeZone = getTimeZone(core.uiSettings);
+
+  const min = Math.min(...xValues);
+  const max = Math.max(
+    ...xValues,
+    // If the last data point of timeseries ends before (picked end time + bucket_span)
+    // we need to extend the x max domain to the last available data point for the expected bounds
+    // for the area chart to show up correctly
+    // See https://github.com/elastic/elastic-charts/issues/1685
+    isComparingExpectedBounds
+      ? anomalyChartTimeseries?.boundaries[0]?.data?.slice(-1)[0]?.x ?? 0
+      : -Infinity
+  );
```

Comment:
Is this related to the sorting as the [linked issue](https://github.com/elastic/elastic-charts/issues/1685) implies?

I don't understand why we need to extend the x-axis range. That should stay true to the time range selected by the user.

Reply:
@sqren This is an unfortunate side effect of elastic-charts. We need to extend the x-axis range here because if not the last portion of the band area will not show up.

So let's say we have an ML expected bound available from 10:30 to 10:45. But if the latest data point of the time series chart is 10:31, as such the x-domain max will be 10:31. If that's the case, the area band will not show up at all for the 10:30 to 10:45 bucket.